### DISCUSSION TITLE
Update of docker-udata dev repository ? still in use ?

### DISCUSSION THREAD

**Discussion by rv2931** [datagouv/data.gouv.fr#1597] [General]
Hello
I'm trying to evaluate the UDATA product based on https://github.com/opendatateam/docker-udata
It looks that this repository is quite out of date (UDATA version 9), docker config (PR: https://github.com/opendatateam/docker-udata/pull/353), it is based on docker image system:py3.11 that is not officially merged in the docker images repositories (https://github.com/opendatateam/udata-dockers/pull/6) so I'm not sure that this is the source repository of this docker image

So my question is: is these repositories docker-udata and udata-dockers still officially and internally used ? or do I have to consider upgrade them on my side (with eventually proposing PRs) ? or should I definitely consider building my own development stack following the native installation documentation ?

I'm interested to evaluate last updates like API automatic generation and restrictive access on ressources

I'm absolutely not against contributing in one way or another and helping to clarify and update certain things but I'm going to need a little help to get started ;)

**Comment by nicolaskempf57** [comment]
Hi and thank you for questions !

If you're interested in trying udata and udata-front, we really recommend [the native installation](https://udata.readthedocs.io/en/stable/getting-started/) because this is the solution we are using and we can easily provide assistance with it.

docker-udata is not a repository internally used so it is not often updated. Please note that if you plan to only use udata and not modify it, you only have to update the requirements.pip to make docker-udata works with the latest udata version.

Thanks for the reminder about https://github.com/opendatateam/udata-dockers/pull/6 being still open. We pushed the docker image based on this branch but never merged it.

Regarding API automatic generation, we are using https://github.com/datagouv/hydra to make the API generation. We were using https://github.com/etalab/csvapi before that and it could be an easier setup depending of your need but will require small adjustments on the udata-front side.

We haven't work on restricted resources yet. We are allowing dataservices (API) with a restricted access to be registered on udata and we will expand the dataset model to do the same thing for resources. The restricted resource metadata will be visible on the dataset page with a link to an authorization request page, but the access in itself will be outside of udata scope.

**Reply by rv2931** [reply]
Thank for your reply
The [docker-udata](https://github.com/opendatateam/docker-udata) is based on published official images
I noticed the [udata-docker-dev-env](https://github.com/opendatateam/udata-docker-dev-env) that intend to supply a docker devlopment stack but it needs git/docker best practice update and looks outdated as it based on https://github.com/etalab/udata-gouvfr this is now archived instead of udata-front

In the other hand the official repository [udata](https://github.com/opendatateam/udata) doesn't supply a docker development stack as many other projects do to ease the onboarding (Linux/Windows). Is this a choice or would you be interested in PR giving a docker stack in udata ? I think I will do it on my side anyway based on the different docker versions available and native installation documentation

**Reply by nicolaskempf57** [reply]
There was several attempts to bring udata to docker so multiple repositories are overlapping on this part but they are mostly left as-is right now. We are archiving udata-docker-dev-env to make it clear that its is not maintained anymore.

If I recall correctly, the main reasons to continue with a native stack was the loss of performance due to Docker Desktop on some OSes and the additional burden required to provide a great developer experience.

The onboarding is quite easy with Python so we never invest too much time on changing it. There is an underlying complexity due to the use of plugins, separate repositories and a shared virtual environment but there is no easy solution to reduce it and its worsen with Docker.

You can make a PR to docker-udata with your udata developer stack if you're willing to so it's available to the community like the [udata-front one](https://github.com/opendatateam/docker-udata/tree/master/samples/theme).
